### ISSUE TITLE
Rule proposal: require node:assert/strict

### ISSUE THREAD

**Issue by electrovir** [sindresorhus/eslint-plugin-unicorn#2420]
### Description

The built-in Node.js `node:assert` module is considered "legacy" and `node:assert/strict` is the non-legacy module. See the docs here: https://nodejs.org/api/assert.html#legacy-assertion-mode.

In particular, those docs note that the legacy `node:assert` module may produce unexpected, rather absurd, results.

### Fail

```js
import assert from 'node:assert'
```

```js
import assert from 'assert'
```

### Pass

```js
import assert from 'node:assert/strict'
```

```js
import assert from 'assert/strict'
```

### Proposed rule name

prefer-node-assert-strict, prefer-strict-node-assert, prefer-strict-node-assert-module, prefer-strict-node-import

**Comment by sindresorhus** [comment]
This would probably be better suited for https://github.com/eslint-community/eslint-plugin-n

**Comment by electrovir** [comment]
Looks like the original `eslint-plugin-node` repo even has a ticket about it: https://github.com/mysticatea/eslint-plugin-node/issues/218

**Comment by electrovir** [comment]
While I don't see a strictly followed conceptual split between this plugin and `eslint-plugin-n`, I can guess where that split would be. If you prefer this proposal to be in `eslint-plugin-n` I can open an issue there.

**Comment by sindresorhus** [comment]
> If you prefer this proposal to be in eslint-plugin-n I can open an issue there.

👍

**Comment by electrovir** [comment]
For reference, I found some relevant tickets already in `eslint-plugin-n`:

- https://github.com/eslint-community/eslint-plugin-n/issues/293
- https://github.com/eslint-community/eslint-plugin-n/issues/59

**Comment by MWhite-22** [comment]
Sorry to bring up an old PR again but I am doing research on eslint plugins and this is one of the only threads I can find talking about unicorn and plugin-n at the same place.

Im doing my best to compare/contrast the two plugins (and a few rules exist across both, like node protocol) but seeing as sindresorhus even calls outs putting this rule into plugin-n, what is the expected split between the two, and is using both a recommendation?

**Comment by fisker** [comment]
> is using both a recommendation

Yes, the two plugins both useful.

> a few rules exist across both, like node protocol

I suggest you can just pick anyone you like, since they are doing the same thing.
But I added support for `process.getBuiltinModule()` recently in #2611, as far as I know `plugin-n` doesn't support it yet.

We already notice the overlap, we may deprecate ours in future, https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1778#issuecomment-2051236918
